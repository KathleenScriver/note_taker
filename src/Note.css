.note {
  border: 2px solid black;
  text-align: center;
  background: #f5f5f5;
  color: #333;
  margin: 20px;
  padding: 20px;
  text-transform: capitalize;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.text {
  flex-basis: 50%;
}

.remove {
  background-color: #9AD4D6;
  border-color: #538083;
  border-width: 1px;
  border-radius: 10px;
}
