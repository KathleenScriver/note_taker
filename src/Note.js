import React, { Component } from 'react';
import './Note.css';

export default class Note extends Component {
  removeNote = () => {
    this.props.removeNote(this.props.name);
  }

  render() {
    return (
      <div className='note'>
        <p>{this.props.name} || {this.props.tag}</p>
        <button className='remove' onClick={this.removeNote}>X</button>
      </div>
    )
  }
}
