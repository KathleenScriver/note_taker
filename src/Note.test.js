import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Note from './Note';

describe(Note, () => {
  const text = 'Get Groceries';
  const tag = 'personal';
  const createdOn = '12/29'
  const mockRemoveNote = jest.fn();
  const component = shallow(
    <Note text={text} tag={tag} createdOn={createdOn} removeNote={mockRemoveNote}/>
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <Note text='Get Groceries' tag='personal' createdOn='12/29'/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('contains the supplied text and tag', () => {
    expect(component.text()).toContain(text);
    expect(component.text()).toContain(tag);
  })

  it('calls passed in removeNote function when remove button clicked', () => {
    component.find('button.remove').simulate('click');
    expect(mockRemoveNote).toBeCalled();
  })
})
