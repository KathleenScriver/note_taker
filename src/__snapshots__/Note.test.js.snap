// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders and matches our snapshot 1`] = `
<div
  className="note"
>
  Get Groceries
  <button
    className="remove"
    onClick={[Function]}
  >
    X
  </button>
</div>
`;
